dev/validate_json.dart

```diff
@@ -3,22 +3,30 @@
 // found in the LICENSE file.
 
 import 'dart:convert';
+import 'dart:io';
 import 'dart:io' show File;
+import 'dart:io' as io_internals show exit;
+
+final bool hasColor = stdout.supportsAnsiEscapes;
+final String bold = hasColor ? '\x1B[1m' : ''; // used for shard titles
+final String red = hasColor ? '\x1B[31m' : ''; // used for errors
+final String reset = hasColor ? '\x1B[0m' : '';
+final String reverse = hasColor ? '\x1B[7m' : ''; // used for clocks
 
 /// Validates if the input builders JSON file has valid contents.
 ///
 /// Examples:
 /// dart validate_json.dart /path/to/json/file
-Future<bool> main(List<String> args) async {
+Future<void> main(List<String> args) async {
   final String jsonString = await File(args[0]).readAsString();
   Map<String, dynamic> decodedJson;
-  bool decodedStatus = true;
+  final List<String> messages = <String>[];
   try {
     decodedJson = json.decode(jsonString) as Map<String, dynamic>;
     final List<dynamic> builders = decodedJson['builders'] as List<dynamic>;
     if (builders == null) {
```

Comment:
The current implementation doesn't check the schema of a builder.

A better way is adding a @JsonSerializable() class representing the schema of the json, then parse the json with that class. We need to parse the json in Cocoon backend code anyway.

Reply:
Good point! Have added logic to check builder schema.